File: src/main/java/com/zunza/buythedip/user/exception/UserNotFoundException.java
package com.zunza.buythedip.user.exception;

import org.springframework.http.HttpStatus;

import com.zunza.buythedip.common.CustomException;

public class UserNotFoundException extends CustomException {
	private final static String MESSAGE = "사용자를 찾을 수 없습니다";

	public UserNotFoundException() {
		super(MESSAGE);
	}

	@Override
	public int getStatusCode() {
		return HttpStatus.NOT_FOUND.value();
	}
}

File: src/main/kotlin/com/zunza/buythedip/user/exception/UserNotFoundException.kt
package com.zunza.buythedip.user.exception

import com.zunza.buythedip.common.CustomException
import org.springframework.http.HttpStatus

class UserNotFoundException(
    id: Long
) : CustomException(MESSAGE + id) {

    companion object {
        private const val MESSAGE = "사용자를 찾을 수 없습니다: "
    }

    override fun getStatusCode() =
        HttpStatus.NOT_FOUND.value()
}
